### PR TITLE
Update release workflow and dependencies

### PR DESCRIPTION
**Pull request by cdr-chakotay**
- Pinned Soatype Staging Profile ID
- Updated Dependencies, in course of that:
  - Removed Test expectation for "Expect: 100-continue" header as [Mockserver does not support it at the moment.](https://github.com/mock-server/mockserver/issues/1151)
  - **Did NOT deactivate sending this header**, as it is needed for the chunked mode.
  - Set min Requirement from Java 6 to Java 7 => Gradle minimum requirements

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -14,20 +14,20 @@ allprojects {
     }
 }
 
-// We compile the library using Java 1.6 compatibility
+// We compile the library using Java 1.7 compatibility
```
Comment (Acconut) [inline build.gradle:17]: Do we need to use 1.7 here now? Is compiling to 1.6 not supported anymore?

File: build.gradle [hunk 2]
```diff
@@ -14,20 +14,20 @@ allprojects {
...
 // in order to ensure interoperability with older Android platforms.
-sourceCompatibility = 1.6
-targetCompatibility = 1.6
+sourceCompatibility = 1.7
+targetCompatibility = 1.7
 
 // load version number from file
-def config = new ConfigSlurper().parse(new File("${projectDir}/src/main/resources/version.properties").toURI().toURL())
+def config = new ConfigSlurper().parse(new File("${projectDir}/src/main/resources/tusVersion/version.properties").toURI().toURL())
```
Comment (Acconut) [inline build.gradle:23]: Can we include the full package name is the path to the version file? e.g.

> resources/io/tus/java/client/version.properties

This way we avoid a name conflict if we also want to have a `version.properties` file in tus-android-client.

File: src/test/java/io/tus/java/client/TestTusUploader.java [hunk 3]
```diff
@@ -69,6 +68,9 @@ public void testTusUploader() throws IOException, ProtocolException {
         assertEquals(-1, uploader.uploadChunk(5));
         assertEquals(11, uploader.getOffset());
         uploader.finish();
+        String log = mockServer.retrieveLogMessages(new HttpRequest());
+        System.out.println("x");
+
```
Comment (Acconut) [inline src/test/java/io/tus/java/client/TestTusUploader.java:73]: These debugging statement can be removed, can't they?

**Review by Acconut — CHANGES_REQUESTED** [review]
Is this ready for review/merge yet?

> Did NOT deactivate sending this header, as it is needed for the chunked mode.

You you explain more why we cannot remove it? Did you try removing it and it caused errors? AFAIK, `connection.setChunkedStreamingMode` should set this header anyways.

File: build.gradle (hunk 1 above)
Comment (cdr-chakotay) [reply build.gradle:17]: Yes Gradle minimal support is now 1.7

File: build.gradle (hunk 2 above)
Comment (cdr-chakotay) [reply build.gradle:23]: of course :)

File: src/test/java/io/tus/java/client/TestTusUploader.java (hunk 3 above)
Comment (cdr-chakotay) [reply src/test/java/io/tus/java/client/TestTusUploader.java:73]: ... yes

**Comment by cdr-chakotay** [comment]
> Is this ready for review/merge yet?
> 
> > Did NOT deactivate sending this header, as it is needed for the chunked mode.
> 
> You you explain more why we cannot remove it? Did you try removing it and it caused errors? AFAIK, `connection.setChunkedStreamingMode` should set this header anyways.

Yes, I can. I stepped through the code and it does not seem to be set. As a result also the Test, which sends a server error 500 to the client does not cause an exception => When I set it manually it waits for the server response

File: build.gradle (hunk 1 above)
Comment (cdr-chakotay) [reply build.gradle:17]: image

**Comment by Acconut** [comment]
> > 
> 
> Yes, I can. I stepped through the code and it does not seem to be set. As a result also the Test, which sends a server error 500 to the client does not cause an exception => When I set it manually it waits for the server response

Ok, interesting. Then let's keep it!

**Review by Acconut — APPROVED** [review]

**Pull request merged**